prefer-destructuring rule falsely reporting array as object when enforceForRenamedProperties is set

**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** v7.7.3
* **npm Version:** v4..1.2

**Please show your full configuration:**
See https://github.com/Shopify/eslint-plugin-shopify
Specifically our [`prefer-destructuring`](https://github.com/Shopify/eslint-plugin-shopify/blob/master/lib/config/rules/ecmascript-6.js#L43) config

```

  'prefer-destructuring': ['error', {array: false, object: true}, {enforceForRenamedProperties: true}],

```

**What did you do? Please include the actual source code causing the issue.**

```js

const arr = [1, 2, 3];
const second = arr[1];
```

**What did you expect to happen?**
No errors, since array has been set to false

**What actually happened? Please include the actual, raw output from ESLint.**
```
  16:9  error    Use object destructuring                     prefer-destructuring
```

--- a/lib/rules/prefer-destructuring.js
+++ b/lib/rules/prefer-destructuring.js
@@ -109,8 +109,10 @@ module.exports = {
                 return;
             }
 
-            if (checkArrays && isArrayIndexAccess(rightNode)) {
-                report(reportNode, "array");
+            if (isArrayIndexAccess(rightNode)) {
+                if (checkArrays) {
+                    report(reportNode, "array");
+                }
                 return;
             }
 

--- a/tests/lib/rules/prefer-destructuring.js
+++ b/tests/lib/rules/prefer-destructuring.js
@@ -72,6 +72,12 @@ ruleTester.run("prefer-destructuring", rule, {
         {
             code: "({ foo } = object);"
         },
+        {
+
+            // Fix #8654
+            code: "var foo = array[0];",
+            options: [{ array: false }, { enforceForRenamedProperties: true }]
+        },
         "[foo] = array;",
         "foo += array[0]",
         "foo += bar.foo"

